protocols/identify/src/lib.rs

```diff
@@ -35,20 +35,29 @@
 //!
 //! # Usage
 //!
-//! The [`Identify`] struct implements a `NetworkBehaviour` that negotiates
+//! The [`Behaviour`] struct implements a `NetworkBehaviour` that negotiates
 //! and executes the protocol on every established connection, emitting
-//! [`IdentifyEvent`]s.
+//! [`Event`]s.
 //!
 //! [Identify]: https://github.com/libp2p/specs/tree/master/identify
-//! [`Identify`]: self::Identify
-//! [`IdentifyEvent`]: self::IdentifyEvent
-//! [`IdentifyInfo`]: self::IdentifyInfo
+//! [`Behaviour`]: self::Behaviour
+//! [`Event`]: self::Event
+//! [`Info`]: self::Info
 
-pub use self::identify::{Identify, IdentifyConfig, IdentifyEvent};
-pub use self::protocol::{IdentifyInfo, UpgradeError, PROTOCOL_NAME, PUSH_PROTOCOL_NAME};
+pub use self::behaviour::{Behaviour, Config, Event};
+pub use self::protocol::{Info, UpgradeError, PROTOCOL_NAME, PUSH_PROTOCOL_NAME};
+
+#[deprecated(
+    since = "0.39.0",
+    note = "Use re-exports that omit `Identify` prefix, i.e. `libp2p::identify::Config` etc"
+)]
+pub use self::behaviour::{Config as IdentifyConfig, Event as IdentifyEvent};
+
+#[deprecated(since = "0.39.0", note = "Use libp2p::identify::Behaviour instead.")]
+pub use Behaviour as Identify;
```

Comment:
I know you copied this from `libp2p-ping` but I just realised that it doesn't actually work :sweat_smile: 
I am not getting any deprecation warnings when I use `IdentifyEvent` and looking through the diff, there are still usages of `PingEvent` that don't get flagged. However, I also just found out that the following works flawlessly:

```suggestion
#[deprecated(
    since = "0.39.0",
    note = "Use re-exports that omit `Identify` prefix, i.e. `libp2p::identify::Config` etc"
)]
pub type IdentifyConfig = Config;

#[deprecated(
    since = "0.39.0",
    note = "Use re-exports that omit `Identify` prefix, i.e. `libp2p::identify::Config` etc"
)]
pub type IdentifyEvent = Event;

#[deprecated(
    since = "0.39.0",
    note = "Use re-exports that omit `Identify` prefix, i.e. `libp2p::identify::Config` etc"
)]
pub type Identify = Behaviour;
```

`note`s could potentially be adjusted to reflect the actual type, I just did it quick and dirty :)

Reply:
ah, interesting! Didn't know that, thanks! :)